homeassistant/components/xiaomi_miio/binary_sensor.py

```diff
@@ -105,6 +106,18 @@ class XiaomiMiioBinarySensorDescription(BinarySensorEntityDescription):
     ),
 }
 
+VACUUM_SENSORS_SEPARATE_MOP = VACUUM_SENSORS | {
+    ATTR_MOP_ATTACHED: XiaomiMiioBinarySensorDescription(
```

Comment:
```suggestion
VACUUM_SENSORS_SEPARATE_MOP = {
    **VACUUM_SENSORS,
    ATTR_MOP_ATTACHED: XiaomiMiioBinarySensorDescription(
```
maybe? Union operator is available only starting from python 3.9.